packages/holocron-dev-server/lib/server.js

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import express from 'express';
+
+import {
+  loadWebpackMiddleware,
+  createRenderingMiddleware,
+  createModulesProxyRelayMiddleware,
+  createMocksMiddleware,
+  requestAcceptedMiddleware,
+} from './middleware';
+import {
+  createModuleMap,
+  getPublicUrl,
+  getStaticPath,
+  loadStatics,
+  loadLanguagePacks,
+  openBrowser,
+} from './utils';
+import {
+  setLogLevel,
+  logHotReloadReady,
+  logServerUrl,
+  logExternalsBundleAnalyzerUrl,
+  logModuleBundlerAnalyzerUrl,
+  logServerStart,
+} from './utils/logs';
+import { errorReportingUrlFragment } from './constants';
+
+export function onLaunch({
+  openWhenReady, externals = [], serverAddress, port,
+}) {
+  return (error) => {
+    if (error) throw error;
+    logServerUrl(serverAddress, port);
+    logHotReloadReady();
+    logModuleBundlerAnalyzerUrl(serverAddress);
+    if (externals.length > 0) {
+      logExternalsBundleAnalyzerUrl(serverAddress, externals);
+    }
+
+    if (openWhenReady) openBrowser(serverAddress);
+  };
+}
+
+export default async function holocronDevServer({
+  port = 4000,
+  serverAddress = `http://localhost:${port}/`,
+  logLevel,
+  clientConfig,
+  openWhenReady,
+  externals,
+  modules,
+  rootModuleName,
+  environmentVariables,
+  remoteModuleMapUrl,
+  dockerImage,
+  webpackConfigPath,
+}) {
+  setLogLevel(logLevel);
```

Comment:
are there no exisiting libraries to manage this ?

Reply:
using something like this https://github.com/pimterry/loglevel? this could be something that we can look to add in subsequent PRs?

Reply:
There is also [`winston`](https://github.com/winstonjs/winston#usage) which can be used as an instance that we configure on start.